### PR TITLE
CTCTRADERS-1498 | Adding Security summary CheckMode navigation and cl…

### PR DESCRIPTION
**Pull request by PSouthern**
…eanup logic

TODO - need to confirm how this should be approached.

```
"must go from CountryOfRouting page to CheckYourAnswers" in {

        navigator
          .nextPage(CountryOfRoutingPage(index), CheckMode, emptyUserAnswers)
          .mustBe(routes.SafetyAndSecurityCheckYourAnswersController.onPageLoad(emptyUserAnswers.id)) //TODO: Possibly change to do below instead
//          .mustBe(routes.AddAnotherCountryOfRoutingController.onPageLoad(emptyUserAnswers.id, CheckMode))
      }
```

### CODE REVIEW

**Comment by PSouthern** [comment]
So I don't forget (following Slack peer review):

- Update nav logic to check if `Yes` route page has previously been answered and redirect accordingly
- Update cleanup specs to add value to cleanup

File: app/navigation/SafetyAndSecurityNavigator.scala [hunk 1]
```diff
@@ -97,25 +121,28 @@ class SafetyAndSecurityNavigator @Inject()() extends Navigator {
...
+      case (Some(true), _) => Some(routes.PlaceOfUnloadingCodeController.onPageLoad(ua.id, mode))
+      case (Some(false), CheckMode) => Some(routes.SafetyAndSecurityCheckYourAnswersController.onPageLoad(ua.id))
+      case (Some(false), NormalMode) =>
         if (totalNumberOfCountriesOfRouting == 0) {
-          routes.CountryOfRoutingController.onPageLoad(ua.id, Index(0), NormalMode)
+          Some(routes.CountryOfRoutingController.onPageLoad(ua.id, Index(0), NormalMode))
         } else {
-          routes.AddAnotherCountryOfRoutingController.onPageLoad(ua.id, NormalMode)
+          Some(routes.AddAnotherCountryOfRoutingController.onPageLoad(ua.id, NormalMode))
         }
```
Comment (kenneth-rayner) [inline app/navigation/SafetyAndSecurityNavigator.scala:136]: In NormalMode will there ever be greater than 0 countries of routing at first should it not always point to  Some(routes.CountryOfRoutingController.onPageLoad(ua.id, Index(0), NormalMode))

File: app/navigation/SafetyAndSecurityNavigator.scala [hunk 2]
```diff
@@ -30,48 +30,70 @@ class SafetyAndSecurityNavigator @Inject()() extends Navigator {
...
+      case (Some(false), NormalMode) => Some(routes.AddCommercialReferenceNumberController.onPageLoad(ua.id, NormalMode))
+      case (Some(false), CheckMode)  => Some(routes.SafetyAndSecurityCheckYourAnswersController.onPageLoad(ua.id))
     }
 
-  def addCommercialReferenceNumber(ua: UserAnswers): Option[Call] =
-    (ua.get(AddCommercialReferenceNumberPage), ua.get(ModeAtBorderPage)) match {
-      case (Some(true), _)                        => Some(routes.AddCommercialReferenceNumberAllItemsController.onPageLoad(ua.id, NormalMode))
-      case (Some(false), Some("4") | Some("40"))  => Some(routes.ConveyanceReferenceNumberController.onPageLoad(ua.id, NormalMode))
-      case (Some(false), _)                       => Some(routes.AddConveyanceReferenceNumberController.onPageLoad(ua.id, NormalMode))
+  def addCommercialReferenceNumber(ua: UserAnswers, mode: Mode): Option[Call] =
```
Comment (kenneth-rayner) [inline app/navigation/SafetyAndSecurityNavigator.scala:80]: make private def

File: app/navigation/SafetyAndSecurityNavigator.scala [hunk 3]
```diff
@@ -97,25 +121,28 @@ class SafetyAndSecurityNavigator @Inject()() extends Navigator {
...
+      case (Some(true), CheckMode) if ua.get(PlaceOfUnloadingCodePage).isDefined => Some(routes.SafetyAndSecurityCheckYourAnswersController.onPageLoad(ua.id))
+      case (Some(true), _) => Some(routes.PlaceOfUnloadingCodeController.onPageLoad(ua.id, mode))
+      case (Some(false), CheckMode) => Some(routes.SafetyAndSecurityCheckYourAnswersController.onPageLoad(ua.id))
+      case (Some(false), NormalMode) =>
         if (totalNumberOfCountriesOfRouting == 0) {
-          routes.CountryOfRoutingController.onPageLoad(ua.id, Index(0), NormalMode)
+          Some(routes.CountryOfRoutingController.onPageLoad(ua.id, Index(0), NormalMode))
         } else {
-          routes.AddAnotherCountryOfRoutingController.onPageLoad(ua.id, NormalMode)
+          Some(routes.AddAnotherCountryOfRoutingController.onPageLoad(ua.id, NormalMode))
```
Comment (kenneth-rayner) [inline app/navigation/SafetyAndSecurityNavigator.scala:135]: In NormalMode will there ever be greater than 0 countries of routing at first should it not always point to Some(routes.CountryOfRoutingController.onPageLoad(ua.id, Index(0), NormalMode))
Would this also apply to def placeOfUnloadingCode ?

Comment (PSouthern) [reply app/navigation/SafetyAndSecurityNavigator.scala:135]: Will leave this as is for now.  May need to be addressed as a follow up with @MatthewAlexanderBeck

File: app/navigation/SafetyAndSecurityNavigator.scala (hunk 2 above)
Comment (PSouthern) [reply app/navigation/SafetyAndSecurityNavigator.scala:80]: 👍

**Review by kenneth-rayner — APPROVED** [review]

**Review by kenneth-rayner — APPROVED** [review]

**Review by kenneth-rayner — APPROVED** [review]

**Review by kenneth-rayner — APPROVED** [review]

**Review by kenneth-rayner — APPROVED** [review]

**Pull request merged**